jolly-jellyfish/src/page_maker/templatetags/render_thumbnail.py

```diff
@@ -0,0 +1,131 @@
+    This is safe to ignore (but can't be escaped, with except, for some reason).
+
+    :param request: A Django HttpRequest object, sourced from the page which is calling this function.
+        This is needed in order to access additional GET info and to build a full url (including domain).
+    :param url_to_render: The relative url of the webpage to render (e.g. '/pages/page_name').
+        Accessed within a template using something like:
+        {% url 'webpage-view' pagename=webpage.name as url_to_render %}.
+        Also needed to build full url.
+    :param page_obj: Either a 'Webpage' or 'Template' Django object. Accessed in order to save thumbnail path.
+    """
+
+    is_template = isinstance(page_obj, Template)  # True if object is an instance of models.Template
+
+    is_outdated = False
+    if not is_template:
+        # only Webpages have the thumbnail_edit_date attr (since they can be edited by the user)
+        # True if the thumbnail is outdated or does not exist yet (thumbnail_edit_date datetime set to 1, 1, 1)
+        is_outdated = page_obj.thumbnail_edit_date < page_obj.last_edit_date
+
+    if is_template:
+        # str(page_obj.thumbnail) returns rel path to thumbnail in str form (not comparable otherwise)
+        if 'placeholder_img.png' not in str(page_obj.thumbnail):  # checks if the template actually needs a thumbnail
+            is_template = False
+
+    # To avoid infinite loops as the page calls itself to render,
+    # ensures function only runs if url doesn't end with '?rendering=true' (added below)
+    is_rendering = bool(request.GET.get('rendering', ''))
+
+    if (is_template or is_outdated) and not is_rendering:  # if a thumbnail is needed
+        base_dir = Path(settings.BASE_DIR)
+        media_dir = Path(settings.MEDIA_ROOT)
+        # 'template_' prefix for Template Objects; 'pg_' for Webpages.
+        image_prefix = 'template_' if is_template else 'pg_'
+        # Can't use only the obj's name as this may contain characters not permitted by file system
+        image_path = Path('thumbnails') / f'{image_prefix}thumb-{page_obj.id}.png'
+        full_os_path = str(base_dir / media_dir / image_path)
+
+        if is_template:
+            window_size = '500,500'  # thumbnails for Templates, due to less content, can be much smaller
+        else:
+            window_size = '1980,1080'
+
+        if 'chromedriver' in DRIVER_PATH:
+            options = add_options(ChromeOptions(), window_size)
+            driver = webdriver.Chrome(executable_path=DRIVER_PATH, options=options)
+        elif 'geckodriver' in DRIVER_PATH:
+            options = add_options(FireOptions(), window_size)
+            driver = webdriver.Firefox(executable_path=DRIVER_PATH, options=options)
+        else:
+            raise Exception('The driver specified in SELENIUM_DRIVER is not supported.\n'
+                            "Currently, only Chrome ('chromedriver') and Firefox ('geckodriver') are supported.\n"
```

Comment:
```suggestion
                            "Currently, only Chrome/Chromium ('chromedriver') and Firefox ('geckodriver') are supported.\n"
```
Chromedriver also works for Chromium.